packages/synapse-react-client/src/components/widgets/facet-nav/SelectionCriteriaPills.tsx

```diff
@@ -88,16 +88,17 @@ function getPillPropsFromColumnQueryFilter(
   })
 }
 
-function getPillPropsFromTextMatchesQueryFilter(
+export function getPillPropsFromTextMatchesQueryFilter(
   queryFilter: TextMatchesQueryFilter,
   queryContext: QueryContextType,
 ): SelectionCriteriaPillProps {
-  const innerText =
-    queryFilter.searchMode == 'NATURAL_LANGUAGE'
-      ? queryFilter.searchExpression
-      : getSearchTextFromBooleanModeSearchExpression(
-          queryFilter.searchExpression,
-        )
```

Comment:
`searchMode` may be undefined. We were improperly using the BOOLEAN logic in that case